pypit/arproc.py

```diff
@@ -857,46 +882,96 @@ def reduce_echelle(slf, sciframe, scidx, fitsdict, det, standard=False):
         slf._bgframe[det - 1] = bgframe
     # Obtain a first estimate of the object trace then
     # fit the traces and perform a PCA for the refinements
-    trccoeff = np.zeros((settings.argflag['trace']['object']['order'], nord))
+    trccoeff = np.zeros((settings.argflag['trace']['object']['order']+1, nord))
     trcxfit = np.arange(nspec)
     for o in range(nord):
         trace, error = artrace.trace_weighted(sciframe-bgframe, slf._lordloc[det-1][:, o], slf._rordloc[det-1][:, o],
                                               mask=slf._scimask[det-1], wght="flux")
+        trace -= slf._lordloc[det-1][:, o]
+        trace /= (slf._rordloc[det-1][:, o]-slf._lordloc[det-1][:, o])
         msk, trccoeff[:, o] = arutils.robust_polyfit(trcxfit, trace,
                                                      settings.argflag['trace']['object']['order'],
                                                      function=settings.argflag['trace']['object']['function'],
                                                      weights=1.0 / error ** 2, minv=0.0, maxv=nspec-1.0)
-    # Identify the orders to be extrapolated during reconstruction
-    orders = 1.0 + np.arange(nord)
-    msgs.info("Performing a PCA on the object trace")
-    ofit = settings.argflag['trace']['object']['pca']['params']
-    lnpc = len(ofit) - 1
-    msgs.work("May need to do a check here to make sure ofit is reasonable")
-    xcen = trcxfit[:, np.newaxis].repeat(nord, axis=1)
-    trccen = arutils.func_val(trccoeff, trcxfit, settings.argflag['trace']['object']['function'],
-                              minv=0.0, maxv=nspec-1.0)
-    fitted, outpar = arpca.basis(xcen, trccen, trccoeff, lnpc, ofit, skipx0=False,
-                                 function=settings.argflag['trace']['object']['function'])
+    refine = 0.0
```

Comment:
I was mainly remarking that this function was getting quite long
so any effort to break it up would be good.